core/shared/src/main/scala/zio/metrics/Metric.scala

```diff
@@ -186,17 +202,43 @@ trait Metric[+Type, -In, +Out] extends ZIOAspect[Nothing, Any, Nothing, Any, Not
 
       override private[zio] val unsafe: UnsafeAPI =
         new UnsafeAPI {
-          override def update(in: In1, extraTags: Set[MetricLabel])(implicit unsafe: Unsafe): Unit =
-            self.unsafe.update(in, f(in) ++ extraTags)
+          override def update(in: In1, extraTags: Set[MetricLabel], description: Option[String])(implicit
+            unsafe: Unsafe
+          ): Unit =
+            self.unsafe.update(in, f(in) ++ extraTags, description)
 
-          override def value(extraTags: Set[MetricLabel])(implicit unsafe: Unsafe): Out =
-            self.unsafe.value(extraTags)
+          override def value(extraTags: Set[MetricLabel], description: Option[String])(implicit unsafe: Unsafe): Out =
+            self.unsafe.value(extraTags, description)
 
-          override def modify(in: In1, extraTags: Set[MetricLabel])(implicit unsafe: Unsafe): Unit =
-            self.unsafe.modify(in, f(in) ++ extraTags)
+          override def modify(in: In1, extraTags: Set[MetricLabel], description: Option[String])(implicit
+            unsafe: Unsafe
+          ): Unit =
+            self.unsafe.modify(in, f(in) ++ extraTags, description)
         }
     }.map(_ => ())
 
+  /**
+   * Returns a new metric, which is identical in every way to this one, except
+   * added description.
+   */
+  final def withDescription(description0: String): Metric[Type, In, Out] = new Metric[Type, In, Out] {
+    override val keyType: Type = self.keyType
+
+    override private[zio] val unsafe: UnsafeAPI = new UnsafeAPI {
+      override def update(in: In, extraTags: Set[MetricLabel], description: Option[String])(implicit
+        unsafe: Unsafe
+      ): Unit =
+        self.unsafe.update(in, extraTags, Some(description.getOrElse(description0)))
```

Comment:
This is a little interesting. The semantics of tags are that adding extra tags always adds tags to the existing tags, whereas this potentially overwrites the description of something that already has a description.